### DISCUSSION TITLE
Headers value in Task

### DISCUSSION THREAD

**Discussion by pratyush-0870** [apache/airflow#28851] [General] [answered]
Hi,

How can I extract the header value which was set in the Airflow API call in Task?

**Comment by Taragolis** [comment, marked as answer]
If I understand you correctly, then you pass some custom headers during REST API call (DagRun or TaskInstance). If so then no, it is not possible because headers not propagated to the tasks.

**Reply by nsAstro** [reply]
Adding to this, if you're trying to trigger a DAGRun via REST and pass information down to the tasks then consider using the 'conf' parameter as specified here: https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run

It takes in a JSON object which can then be accessed by the underlying tasks.

**Reply by pratyush-0870** [reply]
This is my curl request:-
curl --location --request POST 'http://localhost:8080/api/v1/dags/payment_dag/dagRuns' \
--header 'Idempotency-Key: 12345678' \
--header 'Authorization: Basic sdfsdbgkjsfkjdkfjh' \
--header 'Content-Type: application/json' \
--data-raw '{
    "conf": {
        "resources": [
            {
                "funding_account": "1",
                "partner_banks_id": "1",
                "otp_ref_id": "123",
                "payouts": [
                    {
                        "beneficiary": "uuid",
                        "amount": 1,
                        "mode": "imps",
                        "purpose": "testpayment",
                        "udf": {}
                    }
                ]
            }
        ]
    }
}'

I wanted to get 'Idempotency-Key' value from header in Airflow

**Reply by pratyush-0870** [reply]
Thanks for the reply :). I will pass that parameter in conf.
